src/CordovaNativeSqliteProvider.ts

```diff
@@ -138,18 +133,21 @@ export class CordovaNativeSqliteProvider extends SqlProviderBase.SqlProviderBase
 class CordovaNativeSqliteTransaction extends SqlProviderBase.SqliteSqlTransaction {
     constructor(protected trans: SQLTransaction,
                 protected _lockHelper: TransactionLockHelper,
+                protected _transToken: TransactionToken,
                 schema: NoSqlProvider.DbSchema,
-                private _stores: string[],
-                private _exclusive: boolean,
                 verbose: boolean,
                 maxVariables: number,
                 supportsFTS3: boolean) {
         super(trans, schema, verbose, maxVariables, supportsFTS3);
     }
 
-    internal_markTransactionClosed(): void {
-        super.internal_markTransactionClosed();
-        this._lockHelper.transactionComplete(this._stores, this._exclusive);
+    getCompletionPromise(): SyncTasks.Promise<void> {
+        return this._transToken.completionPromise;
+    }
+    
+    abort(): void {
+        this.runQuery('ROLLBACK TRANSACTION');
```

Comment:
this is not going to work.
Will fail as it's done automatically by native layer